gbpservice/neutron/nsf/configurator/agents/generic_config.py

```diff
@@ -0,0 +1,307 @@
+        :param conf: Configuration object that is used for configuration
+        parameter access.
+
+        """
+
+        super(GenericConfigRpcManager, self).__init__(sc, conf)
+
+    def configure_interfaces(self, context, kwargs):
+        """Enqueues event for worker to process configure interfaces request.
+
+        :param context: RPC context
+        :param kwargs: RPC Request data
+
+        Returns: None
+
+        """
+
+        arg_dict = {'context': context,
+                    'kwargs': kwargs}
+        ev = self._sc.event(id='CONFIGURE_INTERFACES', data=arg_dict)
+        self._sc.rpc_event(ev)
+
+    def clear_interfaces(self, context, kwargs):
+        """Enqueues event for worker to process clear interfaces request.
+
+        :param context: RPC context
+        :param kwargs: RPC Request data
+
+        Returns: None
+
+        """
+
+        arg_dict = {'context': context,
+                    'kwargs': kwargs}
+        ev = self._sc.event(id='CLEAR_INTERFACES', data=arg_dict)
+        self._sc.rpc_event(ev)
+
+    def configure_routes(self, context, kwargs):
+        """Enqueues event for worker to process configure routes request.
+
+        :param context: RPC context
+        :param kwargs: RPC Request data
+
+        Returns: None
+
+        """
+
+        arg_dict = {'context': context,
+                    'kwargs': kwargs}
+        ev = self._sc.event(id='CONFIGURE_ROUTES', data=arg_dict)
+        self._sc.rpc_event(ev)
+
+    def clear_routes(self, context, kwargs):
+        """Enqueues event for worker to process clear routes request.
+
+        :param context: RPC context
+        :param kwargs: RPC Request data
+
+        Returns: None
+
+        """
+
+        arg_dict = {'context': context,
+                    'kwargs': kwargs}
+        ev = self._sc.event(id='CLEAR_ROUTES', data=arg_dict)
+        self._sc.rpc_event(ev)
+
+"""Implements event handlers and their helper methods.
+
+Object of this class is registered with the event class of core service
+controller. Based on the event key, handle_event method of this class is
+invoked by core service controller.
+
+"""
+
+
+class GenericConfigEventHandler(agent_base.AgentBaseEventHandler):
+    def __init__(self, sc, drivers, rpcmgr, nqueue):
+        super(GenericConfigEventHandler, self).__init__(
+                                        sc, drivers, rpcmgr, nqueue)
+
+    def _get_driver(self, service_type):
+        """Retrieves service driver object based on service type input.
+
+        Currently, service drivers are identified with service type. Support
+        for single driver per service type is provided. When multi-vendor
+        support is going to be provided, the driver should be selected based
+        on both service type and vendor name.
+
+        :param service_type: Service type - firewall/vpn/loadbalancer
+
+        Returns: Service driver instance
+
+        """
+
+        return self.drivers[service_type]()
```

Comment:
Yes. We're creating a driver object here

Regards,
Deepak

On Tue, Feb 23, 2016 at 12:46 PM, pkharat <notifications@github.com> wrote:

> In gbpservice/neutron/nsf/configurator/agents/generic_config.py
> <https://github.com/oneconvergence/group-based-policy/pull/59#discussion_r53743481>
> :
>
> > +
> > +    def _get_driver(self, service_type):
> > +        """Retrieves service driver object based on service type input.
> > +
> > +        Currently, service drivers are identified with service type. Support
> > +        for single driver per service type is provided. When multi-vendor
> > +        support is going to be provided, the driver should be selected based
> > +        on both service type and vendor name.
> > +
> > +        :param service_type: Service type - firewall/vpn/loadbalancer
> > +
> > +        Returns: Service driver instance
> > +
> > +        """
> > +
> > +        return self.drivers[service_type]()
>
> Are we creating new driver object here ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/oneconvergence/group-based-policy/pull/59/files#r53743481>
> .
>